apps/s_server.c

```diff
@@ -2626,6 +2626,10 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
                 if (buf[0] == 'S') {
                     print_stats(bio_s_out, SSL_get_SSL_CTX(con));
                 }
+            } else if (i <= 0) {
+                (void)BIO_flush(bio_s_out);
```

Comment:
In the not-quiet case we do this:

````
                    (void)BIO_flush(bio_s_out);
                    BIO_closesocket(s);
                    close_accept_socket();
                    ret = -11;
                    goto err;
````

Shouldn't the behaviour be the same?

Reply:
In the current situation, the quiet mode and the non-quiet mode are inconsistent. In the non-quiet mode, EOF is treated as the same effect as the input `Q`. End the current connection and exit listen socket. In the quiet mode, will get the error result after calling `SSL_write()`, and then the current ssl connection is ended, and the listen socket is still keeped. This patch only checks whether it is EOF before calling `SSL_write()`, so as to avoid calling SSL_write() and then making an error.
At present, the process in non-quiet and quiet mode is inconsistent. I think there is no problem, and it is also in line with the usage habits.